android/src/main/java/com/cubicphuse/RCTTorch/RCTTorchModule.java

```diff
@@ -49,14 +49,14 @@ public void switchState(Boolean newState, Callback successCallback, Callback fai
         } else {
             Camera.Parameters params;
 
-            if (!isTorchOn) {
+            if (newState && !isTorchOn) {
```

Comment:
It's been a while since I've looked at this code so I'm trying to wrap my head around it again. Could we do away with `isTorchOn` altogether and just use `if (newState) { ... } else { ... }`?

Reply:
actually no, I've tried to do this - but there is an error in this case. You can check it by yourself :) Probably, as a condition handler we have specific code which fails in case we try to set 'ON' state on already 'ON' state set 
![Skype](https://user-images.githubusercontent.com/19843391/56292948-d0becb00-6130-11e9-9758-66c04b0a2d15.png)